api/coreservice_test.go

```diff
@@ -7,16 +7,139 @@
+		require.NoError(err)
+		require.Equal(0, len(logs))
+	})
+	t.Run("over 5000 pagenation size", func(t *testing.T) {
+		testData := &filterObject{FromBlock: "1", ToBlock: "4"}
+		filter, err := getTopicsAddress(testData.Address, testData.Topics)
+		require.NoError(err)
+		from, err := strconv.ParseUint(testData.FromBlock, 10, 64)
+		require.NoError(err)
+		to, err := strconv.ParseUint(testData.ToBlock, 10, 64)
+		require.NoError(err)
+
+		logs, err := svr.web3Server.coreService.LogsInRange(logfilter.NewLogFilter(&filter, nil, nil), from, to, uint64(5001))
+		require.NoError(err)
+		require.Equal(4, len(logs))
+	})
+	t.Run("invalid start and end height", func(t *testing.T) {
+		testData := &filterObject{FromBlock: "2", ToBlock: "1"}
+		filter, err := getTopicsAddress(testData.Address, testData.Topics)
+		require.NoError(err)
+		from, err := strconv.ParseUint(testData.FromBlock, 10, 64)
+		require.NoError(err)
+		to, err := strconv.ParseUint(testData.ToBlock, 10, 64)
+		require.NoError(err)
+
+		logs, err := svr.web3Server.coreService.LogsInRange(logfilter.NewLogFilter(&filter, nil, nil), from, to, uint64(0))
+		expectedErr := errors.New("invalid start and end height")
+		expectedValue := []*iotextypes.Log(nil)
+		require.Error(err)
+		require.Equal(expectedErr.Error(), err.Error())
+		require.Equal(expectedValue, logs)
+	})
+	t.Run("start block > tip height", func(t *testing.T) {
+		testData := &filterObject{FromBlock: "5", ToBlock: "5"}
+		filter, err := getTopicsAddress(testData.Address, testData.Topics)
+		require.NoError(err)
+		from, err := strconv.ParseUint(testData.FromBlock, 10, 64)
+		require.NoError(err)
+		to, err := strconv.ParseUint(testData.ToBlock, 10, 64)
+		require.NoError(err)
+
+		logs, err := svr.web3Server.coreService.LogsInRange(logfilter.NewLogFilter(&filter, nil, nil), from, to, uint64(0))
+		expectedErr := errors.New("start block > tip height")
+		expectedValue := []*iotextypes.Log(nil)
+		require.Error(err)
+		require.Equal(expectedErr.Error(), err.Error())
+		require.Equal(expectedValue, logs)
+	})
+}
+
+func BenchmarkLogsInRange(b *testing.B) {
+	svr, _, _, _, cleanCallback := setupTestServer()
+	defer cleanCallback()
+
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+	core := mock_apicoreservice.NewMockCoreService(ctrl)
+	
+	testData := &filterObject{FromBlock: "0x1"}
+	filter, _ := getTopicsAddress(testData.Address, testData.Topics)
+	from, _ := strconv.ParseInt(testData.FromBlock, 10, 64)
+	to, _ := strconv.ParseInt(testData.ToBlock, 10, 64)
+
+	b.Run("five workers to extract logs", func(b *testing.B) {
+		core.EXPECT().LogsInRange(logfilter.NewLogFilter(&filter, nil, nil), uint64(from), uint64(to), uint64(0)).DoAndReturn(func(filter *logfilter.LogFilter, start, end, paginationSize uint64) ([]*iotextypes.Log, error) {
+			logs, _ := svr.web3Server.coreService.LogsInRange(filter, start, end, paginationSize)
```

Comment:
This bottleneck is at
`blockNumbers, err := core.bfIndexer.FilterBlocksInRange(filter, start, end)`, which should be mocked.

Reply:
That means I need to generate a new mock `FilterBlocksInRange`?